doc/Data Plane Code Generation/SubClients.md

```diff
@@ -0,0 +1,178 @@
+# Customizing the generated code
+
+## SubClients
+
+There are two categories of clients: service clients and their subclients. Service clients can be instantiated and have the Client suffix. Subclients can only be created by calling factory methods on other clients (commonly on service clients) and do not have the client suffix. The service client is the entry point to the API for an Azure service.
+
+There are two approaches to organize subclients and service client.
```

Comment:
Maybe if we said instead of "two approaches" that we give you two tools to configure how your service clients and subclients are related: the first lets you specify which is the top-level client, and the other lets you specify parent-child relationships between the subclients.  The latter allows you to make the client hierarchy deeper.  You can use them together.  Suggestions for when to add a subclient are here: https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-choosing-client-types

Reply:
updated, please help to review.